dulwich/porcelain.py

```diff
@@ -408,6 +408,54 @@ def add(repo=".", paths=None):
     return (relpaths, ignored)
 
 
+def is_subdir(subdir, parentdir):
```

Comment:
Please make is_subdir private, especially since it doesn't have its own unit tests.